doc/rtd/explanation/hardening.rst

```diff
@@ -0,0 +1,18 @@
+Security Hardening
+******************
+
+Cloud-init's use case is automating cloud instance initialization, with support
+across distributions and platforms. There is a myriad of ways to harden this
+space.
+
+Follow the security hardening guidelines provided by the OSes and cloud
+platforms that your cloud-init configuration is targeting.
+
+No plain text passwords
+=======================
+
+While creating users with the
+:ref:`Users and Groups module <mod_cc_users_groups>`, do not use the
+`user.plain_text_passwd` key with its associated value as plain text.
+As anyone with access to the configuration might have access to the password.
```

Comment:
Another suggestion for hardening security of launched VMs in cloud-init is to document that cloud-init publishes the SSH host public key generated to the serial console which can be validated prior to any SSH client connection to the launched VM. It's give a sense of security that you are actually attaching to the VM you thought you launched and not a MITM.